framework/provider/provider_configure.go

```diff
@@ -0,0 +1,160 @@
+		for _, p := range configPaths {
+			path, err := homedir.Expand(p)
+			if err != nil {
+				return nil, err
+			}
+
+			tflog.Debug(ctx, "Using kubeconfig file", map[string]interface{}{"path": path})
+			expandedPaths = append(expandedPaths, path)
+		}
+		if len(expandedPaths) == 1 {
+			loader.ExplicitPath = expandedPaths[0]
+		} else {
+			loader.Precedence = expandedPaths
+		}
+
+		ctxSuffix := "; default context"
+
+		kubectx := data.ConfigContext.ValueString()
+		authInfo := data.ConfigContextAuthInfo.ValueString()
+		cluster := data.ConfigContextCluster.ValueString()
+		if kubectx != "" || authInfo != "" || cluster != "" {
+			ctxSuffix = "; overridden context"
+			if kubectx != "" {
+				overrides.CurrentContext = kubectx
+				ctxSuffix += fmt.Sprintf("; config ctx: %s", overrides.CurrentContext)
+				tflog.Debug(ctx, "Using custom current context", map[string]interface{}{"context": overrides.CurrentContext})
+			}
+
+			overrides.Context = clientcmdapi.Context{}
+			if authInfo != "" {
+				overrides.Context.AuthInfo = authInfo
+				ctxSuffix += fmt.Sprintf("; auth_info: %s", overrides.Context.AuthInfo)
+			}
+			if cluster != "" {
+				overrides.Context.Cluster = cluster
+				ctxSuffix += fmt.Sprintf("; cluster: %s", overrides.Context.Cluster)
+			}
+			tflog.Debug(ctx, "Using overridden context", map[string]interface{}{"context": overrides.Context})
+		}
+	}
+
+	// Overriding with static configuration
+	overrides.ClusterInfo.InsecureSkipTLSVerify = data.Insecure.ValueBool()
+	overrides.ClusterInfo.TLSServerName = data.TLSServerName.ValueString()
+	overrides.ClusterInfo.CertificateAuthorityData = bytes.NewBufferString(data.ClusterCACertificate.ValueString()).Bytes()
+	overrides.AuthInfo.ClientCertificateData = bytes.NewBufferString(data.ClientCertificate.ValueString()).Bytes()
+
+	if v := data.Host.ValueString(); v != "" {
+		// Server has to be the complete address of the kubernetes cluster (scheme://hostname:port), not just the hostname,
+		// because `overrides` are processed too late to be taken into account by `defaultServerUrlFor()`.
+		// This basically replicates what defaultServerUrlFor() does with config but for overrides,
+		// see https://github.com/kubernetes/client-go/blob/v12.0.0/rest/url_utils.go#L85-L87
+		hasCA := len(overrides.ClusterInfo.CertificateAuthorityData) != 0
+		hasCert := len(overrides.AuthInfo.ClientCertificateData) != 0
+		defaultTLS := hasCA || hasCert || overrides.ClusterInfo.InsecureSkipTLSVerify
+		host, _, err := restclient.DefaultServerURL(v, "", apimachineryschema.GroupVersion{}, defaultTLS)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse host: %s", err)
+		}
+
+		overrides.ClusterInfo.Server = host.String()
+	}
+
+	overrides.AuthInfo.Username = data.Username.ValueString()
+	overrides.AuthInfo.Password = data.Password.ValueString()
+	overrides.AuthInfo.ClientKeyData = bytes.NewBufferString(data.ClientKey.ValueString()).Bytes()
+	overrides.AuthInfo.Token = data.Token.ValueString()
```

Comment:
If this value was Null or Unknown (e.g. in a progressive apply situation where provider config values aren't known yet) the strings here, I think, would be `<null>` or `<unknown>` That might not cause any problems though, depending on how the overrides are used.

Reply:
The godoc for this function says:

```
ValueString returns the known string value. If String is null or unknown, returns "".
```

So this is fine, as all of these string fields on the override struct would default to `""` anyway.